File: library/src/main/java/com/nshmura/strictmodenotifier/StrictModeNotifier.java
package com.nshmura.strictmodenotifier;

import android.content.Context;

public class StrictModeNotifier {

  public static NotifierConfig install(Context context) {
    return install(context, LogWatchService.class);
  }

  public static NotifierConfig install(Context context,
      Class<? extends LogWatchService> serviceClass) {
    StrictModeNotifierInternals.enableReportActivity(context);
    StrictModeNotifierInternals.startLogWatchService(context, serviceClass);
    return NotifierConfig.getInstance();
  }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/StrictModeNotifier.kt
package io.github.juanimoli.strictmode.notifier

import android.content.Context
import io.github.juanimoli.strictmode.notifier.commons.NotifierConfig

object StrictModeNotifier {
    @JvmOverloads
    fun install(
        context: Context,
        serviceClass: Class<out LogWatchService?>? = LogWatchService::class.java
    ): NotifierConfig {
        StrictModeNotifierInternals.enableReportActivity(context)
        StrictModeNotifierInternals.startLogWatchService(context, serviceClass)

        return NotifierConfig.instance
    }
}
